### PR TITLE
fix: add flash to scanner and improve ui

### PR DESCRIPTION
**Pull request by SputNikPlop**
This was it before
IMG_2188 Medium

and now after
IMG_2187 Medium

### CODE REVIEW

File: lib/screens/settings/audio_sources.dart [hunk 1]
```diff
@@ -141,42 +143,122 @@ class _AudioSourcesScreenState extends State<AudioSourcesScreen> {
...
+                                                        IconButton(
+                                                            onPressed: () =>
+                                                                _scanController
+                                                                    .toggleTorch(),
+                                                            icon: icon),
+                                                      ],
+                                                    );
+                                                  },
+                                                ),
+                                              ),
```
Comment (kevmo314) [inline lib/screens/settings/audio_sources.dart:248]: Move this to a separate widget so it can be shared between the two scanner implementations.

Comment (SputNikPlop) [reply lib/screens/settings/audio_sources.dart]: Ok this makes more sense now.

File: lib/screens/settings/activity_feed.dart [hunk 2]
```diff
@@ -151,42 +152,112 @@ class _ActivityFeedScreenState extends State<ActivityFeedScreen> {
...
-                                    },
+                                          Navigator.pop(ctx);
+                                        },
+                                      ),
+                                      Positioned(
+                                        top: 50,
+                                        left: 0,
+                                        right: 0,
+                                        child:
+                                            ValueListenableBuilder<TorchState>(
```
Comment (kevmo314) [inline lib/screens/settings/activity_feed.dart:202]: Replace these too

**Review by kevmo314 — CHANGES_REQUESTED** [review]

**Comment by SputNikPlop** [comment]
@kevmo314 Do I put the scanner_settings as a component or in the model? I don't think I should keep it under screens

File: lib/screens/settings/activity_feed.dart (hunk 2 above)
Comment (SputNikPlop) [reply lib/screens/settings/activity_feed.dart]: Done

File: lib/screens/settings/scanner_settings.dart [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'package:mobile_scanner/mobile_scanner.dart';
+
+Widget scannerSettings(
```
Comment (kevmo314) [inline lib/screens/settings/scanner_settings.dart:4]: This should be a full widget, not just a function that returns one.

```
class ScannerSettings extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    ...
  }
}
```

File: lib/components/scanner_settings.dart [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+              color: iconColor,
+            );
+            break;
+          case TorchState.off:
+            icon = const Icon(
+              Icons.flash_on,
+              color: iconColor,
+            );
+            break;
+        }
```
Comment (kevmo314) [inline lib/components/scanner_settings.dart:30]: Inline this into the `const Icon()` below:

```
icon: Icon(
  value == TorchState.on ? Icons.flash_off : Icons.flash_on,
  color: iconColor,
),
```

**Review by kevmo314 — APPROVED** [review]

**Pull request merged**